datafusion/optimizer/src/extract_equijoin_predicate.rs

```diff
@@ -28,7 +28,17 @@ use std::sync::Arc;
 // equijoin predicate
 type EquijoinPredicate = (Expr, Expr);
 
-/// Optimization rule that extract equijoin expr from the filter
+/// Optimizer that splits conjunctive join predicates into equijoin
+/// predicates and filter (other) predicates.
```

Comment:
Small suggestion. It reads more smooth to me:

```suggestion
/// Optimizer that splits conjunctive join predicates into equijoin
/// predicates and (other) filter predicates.
```